dear_petition/petition/api/viewsets.py

```diff
@@ -149,6 +148,22 @@ class ContactViewSet(viewsets.ModelViewSet):
     ordering_fields = ["name", "address1", "address2", "city", "zipcode"]
     ordering = ["name"]
 
+    def get_serializer_class(self):
+        if self.request.data.get('category') == 'client':
+            return serializers.ClientSerializer
+        return serializers.ContactSerializer
```

Comment:
Maybe if client data and contact data diverge in the future, we might want to look into separating them into 2 tables.